### PR TITLE
Rebuilt missing functionalities, some tweaking still needed

### PR DESCRIPTION
**Pull request by Ilithor**

### CODE REVIEW

File: src/components/PostContent.js [hunk 1]
```diff
@@ -16,7 +17,7 @@ import ChatIcon from '@material-ui/icons/Chat';
 /** View component for displaying an individual post's content'
  * @param {IPostComponentProps} props
  */
```
Comment (jeffersoneagley) [inline src/components/PostContent.js]: add props.post and props.classes to your type annotation here

File: src/components/UserProfileDisplay.js [hunk 2]
```diff
@@ -115,17 +117,18 @@ class UserProfileDisplay extends Component {
   }
 }
 
-const mapStateToProps = state => {
-  return {
-    user: state.user,
-  };
-};
+const mapStateToProps = state => ({
```
Comment (jeffersoneagley) [inline src/components/UserProfileDisplay.js]: noice

File: src/pages/home.js [hunk 3]
```diff
@@ -1,25 +1,21 @@
...
-        postList: res.data,
-      });
-    });
+    this.props.getPostList();
   }
   render() {
-    let recentPostMarkup = this.state.postList ? (
-      this.state.postList.map(post => <Post key={post.postId} post={post} />)
+    const { postList, isLoading } = this.props.data;
+    let recentPostMarkup = !isLoading ? (
```
Comment (jeffersoneagley) [inline src/pages/home.js]: Too long for a ternary operator
if(isLoading){
return <p>Loading. . .</p>
}
//rest of stuff

Comment (Ilithor) [reply src/pages/home.js]: Fixed

File: src/components/Navbar.js [hunk 4]
```diff
@@ -121,7 +121,9 @@ Navbar.propTypes = {
   logoutUserAction: PropTypes.func,
 };
 
-const mapStateToProps = ({ user }) => ({ isLoggedIn: !!user && !!user.handle });
+const mapStateToProps = ({ user }) => ({
+  isLoggedIn: !!user && !!user.userInfo.handle,
```
[the remark this answers is not in the document]
Comment (Ilithor) [reply src/components/Navbar.js]: Fixed

File: src/components/PostContent.js (hunk 1 above)
Comment (Ilithor) [reply src/components/PostContent.js]: Fixed

File: src/components/Navbar.js [hunk 5]
```diff
@@ -121,7 +121,12 @@ Navbar.propTypes = {
   logoutUserAction: PropTypes.func,
 };
 
-const mapStateToProps = ({ user }) => ({ isLoggedIn: !!user && !!user.handle });
+const mapStateToProps = ({ user }) => {
+  const isLoggedIn = !!user && !!user.userInfo.handle;
```
Comment (jeffersoneagley) [inline src/components/Navbar.js]: you're not null-checking userInfo here
add `&& !!user.userinfo`

Comment (Ilithor) [reply src/components/Navbar.js]: Fixed

**Pull request merged**